ibis/backends/snowflake/__init__.py

```diff
@@ -973,6 +973,45 @@ def read_parquet(
 
         return self.table(table)
 
+    def insert(
+        self,
+        table_name: str,
+        obj: pd.DataFrame | ir.Table | list | dict,
+        schema: str | None = None,
+        database: str | None = None,
+        overwrite: bool = False,
+    ) -> None:
+        """Insert data into a table.
+
+        Parameters
+        ----------
+        table_name
+            The name of the table to which data needs will be inserted
+        obj
+            The source data or expression to insert
+        schema
+            The name of the schema that the table is located in
+        database
+            Name of the attached database that the table is located in.
+        overwrite
+            If `True` then replace existing contents of table
+        """
+        if not isinstance(obj, ir.Table):
+            obj = ibis.memtable(obj)
+
+        self._run_pre_execute_hooks(obj)
+        query = sg.exp.insert(
+            expression=self.compile(obj),
+            into=sg.table(table_name, db=schema, catalog=database, quoted=True),
+            columns=[sg.column(col, quoted=True) for col in obj.columns],
+            dialect=self.name,
+        )
+        with self.begin() as con:
+            if overwrite:
```

Comment:
We can't pass this to `insert` because of https://github.com/tobymao/sqlglot/issues/2815